master/internal/api.go

```diff
@@ -159,18 +159,28 @@ func (a *apiServer) filter(values interface{}, check func(int) bool) {
 	rv.Elem().Set(results)
 }
 
-func (a *apiServer) actorRequest(addr string, req actor.Message, v interface{}) error {
+func (a *apiServer) actorRequestExpanded(addr string, req actor.Message, v interface{}) (
+	codes.Code, error) {
 	actorAddr := actor.Address{}
 	if err := actorAddr.UnmarshalText([]byte(addr)); err != nil {
-		return status.Errorf(codes.InvalidArgument, "/api/v1%s is not a valid path", addr)
+		code := codes.InvalidArgument
+		message := fmt.Sprintf("/api/v1%s is not a valid path", addr)
+		return code, status.Error(code, message)
 	}
 	resp := a.m.system.AskAt(actorAddr, req)
 	if resp.Empty() {
-		return status.Errorf(codes.NotFound, "/api/v1%s not found", addr)
+		code := codes.NotFound
+		message := fmt.Sprintf("/api/v1%s not found", addr)
+		return code, status.Error(code, message)
 	}
 	if err := resp.Error(); err != nil {
-		return err
+		return codes.Internal, err
 	}
 	reflect.ValueOf(v).Elem().Set(reflect.ValueOf(resp.Get()))
-	return resp.Error()
+	return codes.Internal, resp.Error()
+}
+
+func (a *apiServer) actorRequest(addr string, req actor.Message, v interface{}) error {
```

Comment:
blocking: You should get the code from the error instead of actorRequestExpanded (you should check the type and see that it is a status error).

Reply:
the error type only exposes `Error() string` should I be checking it against a different error type?

Reply:
update: using status.Code() now